seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/OrcWriteStrategy.java

```diff
@@ -0,0 +1,210 @@
+                return newWriter;
+            } catch (IOException e) {
+                String errorMsg = String.format("Get orc writer for file [%s] error", filePath);
+                throw new RuntimeException(errorMsg, e);
+            }
+        }
+        return writer;
+    }
+
+    private TypeDescription buildFieldWithRowType(SeaTunnelDataType<?> type) {
+        if (BasicType.BOOLEAN_TYPE.equals(type)) {
+            return TypeDescription.createBoolean();
+        }
+        if (BasicType.SHORT_TYPE.equals(type)) {
+            return TypeDescription.createShort();
+        }
+        if (BasicType.INT_TYPE.equals(type)) {
+            return TypeDescription.createInt();
+        }
+        if (BasicType.LONG_TYPE.equals(type)) {
+            return TypeDescription.createLong();
+        }
+        if (BasicType.FLOAT_TYPE.equals(type)) {
+            return TypeDescription.createFloat();
+        }
+        if (BasicType.DOUBLE_TYPE.equals(type)) {
+            return TypeDescription.createDouble();
+        }
+        if (BasicType.BYTE_TYPE.equals(type)) {
+            return TypeDescription.createByte();
+        }
+        return TypeDescription.createString();
+    }
+
+    private TypeDescription buildSchemaWithRowType() {
+        TypeDescription schema = TypeDescription.createStruct();
+        for (Integer i : sinkColumnsIndexInRow) {
+            TypeDescription fieldType = buildFieldWithRowType(seaTunnelRowType.getFieldType(i));
+            schema.addField(seaTunnelRowType.getFieldName(i), fieldType);
+        }
+        return schema;
+    }
+
+    private void setColumn(Object value, ColumnVector vector, int row) {
+        if (value == null) {
+            vector.isNull[row] = true;
+            vector.noNulls = false;
+        } else {
+            switch (vector.type) {
+                case LONG:
+                    LongColumnVector longVector = (LongColumnVector) vector;
+                    setLongColumnVector(value, longVector, row);
+                    break;
+                case DOUBLE:
+                    DoubleColumnVector doubleColumnVector = (DoubleColumnVector) vector;
+                    setDoubleVector(value, doubleColumnVector, row);
+                    break;
+                case BYTES:
+                    BytesColumnVector bytesColumnVector = (BytesColumnVector) vector;
+                    setByteColumnVector(value, bytesColumnVector, row);
+                    break;
+                default:
+                    throw new RuntimeException("Unexpected ColumnVector subtype");
+            }
+        }
+    }
+
+    private void setLongColumnVector(Object value, LongColumnVector longVector, int row) {
+        if (value instanceof Boolean) {
+            Boolean bool = (Boolean) value;
+            longVector.vector[row] = (bool.equals(Boolean.TRUE)) ? Long.valueOf(1) : Long.valueOf(0);
+        }  else if (value instanceof Integer) {
+            longVector.vector[row] = (Integer) value;
```

Comment:
`(Integer) value` -> `((Long) value).longValue()`

Reply:
The object that can enter the judgment logic has been determined to be Integer, is it necessary to force the conversion to Long?

Reply:
The target type of this method is Long type, use `LongColumnVector.vector[i] = (Long)x` is better than `LongColumnVector.vector[i] = (Integer)x`

Reply:
Got it. Thank you for your good advice.